Tests/Form/Type/SimpleMoneyTypeTest.php

```diff
@@ -72,12 +62,33 @@ public function testGreaterThan1000Valid()
     public function testSetData()
     {
         \Locale::setDefault("fr_FR");
-        $moneyType = new SimpleMoneyType($this->pairManager, 2);
-        $form = $this->factory->create($moneyType, null, array());
+        $form = $this->factory->create($this->simpleMoneyTypeClass, null, array());
         $form->setData(Money::EUR(120));
         $formView = $form->createView();
 
         $this->assertEquals("1,20", $formView->children["tbbc_amount"]->vars["value"]);
     }
 
+    protected function getExtensions()
+    {
+        //This is probably not ideal, but I'm not sure how to set up the pair manager
+        // with different decimals for different tests in Symfony 3.0
```

Comment:
@drewclauson No worries, you made some great work!

Here's an example of how it works for  CurrencyType:

```
$form = $this->factory->create('Tbbc\MoneyBundle\Form\Type\CurrencyType', null, array(
    'reference_currency' => 'EUR',
    'currency_choices' => ['EUR','USD']
));
```

So as you can see you provide optional details as third parameter of the factory's `create` method. These parameters are accepted because of these definitions in `CurrencyType`:

```
        $resolver->setDefaults(array(
            'reference_currency' => $this->referenceCurrencyCode,
            'currency_choices' => $this->currencyCodeList
        ));
        $resolver->setAllowedTypes('reference_currency', 'string');
        $resolver->setAllowedTypes('currency_choices', 'array');
```

For `MoneyType` however it might be more difficult after symfony changes. Because currently it expects preconfigured `CurrencyType` object as an argument, which we cannot do with symfony 3. So I think we maybe need to have `reference_currency` and `currency_choices` on `MoneyType` as well and then pass it on to `CurrencyType`.

Thoughts @drewclauson @philippe-levan?